src/wasm/wasm-binary.cpp

```diff
@@ -317,9 +317,14 @@ void WasmBinaryWriter::writeDataCount() {
   if (!wasm->features.hasBulkMemory() || !wasm->memory.segments.size()) {
     return;
   }
-  auto start = startSection(BinaryConsts::Section::DataCount);
-  o << U32LEB(wasm->memory.segments.size());
-  finishSection(start);
+
+  // TODO(tlively): re-enable writing the data count once the default feature
+  // set is no longer All, which causes validation errors in Emscripten due to
+  // the presence of an unrecognized section.
```

Comment:
I don't think that's the issue - emscripten works ok, but when node tries to run it, it throws `CompileError: AsyncCompile: Wasm decoding failed: unexpected section: Code @+243`.

So the problem seems to be that we are emitting that section when we should not be?

Anyhow, lgtm to land this if it unbreaks things for now.

Reply:
hmmm, I should probably figure out what's actually happening then.